src/prog_models/prognostics_model.py

```diff
@@ -1,23 +1,43 @@
-# Copyright © 2021 United States Government as represented by the Administrator of the National Aeronautics and Space Administration.  All Rights Reserved.
+# Copyright © 2021 United States Government as represented by the Administrator of the
+# National Aeronautics and Space Administration.  All Rights Reserved.
 
 from .exceptions import ProgModelInputException, ProgModelTypeError, ProgModelException
 from abc import abstractmethod, ABC
 from numbers import Number
-from numpy import random
+import numpy as np
 from copy import deepcopy
 from collections import UserDict
 import types
+from array import array
+from .sim_result import SimResult, LazySimResult
+
 
 class PrognosticsModelParameters(UserDict):
     """
-    Prognostics Model Parameters - this class replaces a standard dictionary. 
+    Prognostics Model Parameters - this class replaces a standard dictionary.
     It includes the extra logic to process the different supported manners of defining noise.
+
+    Args:
+        model: PrognosticsModel for which the params correspond
+        dict_in: Initial parameters
+        callbacks: Any callbacks for derived parameters f(parameters) : updates (dict)
     """
-    def __init__(self, model, dict_in = {}):
+    def __init__(self, model, dict_in = {}, callbacks = {}):
         super().__init__()
         self.__m = model
+        self.callbacks = {}
```

Comment:
The need for this line is not clear because line 35 resets the classes' callbacks (defaulting to {}). We did a small test and it doesn't seem to work without it though. This is more of a curiosity for us than a bug/fix.

Reply:
This is needed to prevent the initial setting on 31 from calling the callbacks. This could call a callback which uses a parameter that hasn't yet been set, depending on parameter order as it iterates through dict_in on line 29.

Reply:
I added a comment there to explain for other users. Excellent question!